lib/explorer/polars_backend/data_frame.ex

```diff
@@ -753,6 +753,26 @@ defmodule Explorer.PolarsBackend.DataFrame do
     Shared.apply_dataframe(df, out_df, :df_explode, [columns])
   end
 
+  @impl true
+  def correlation(df, out_df, ddof) do
+    [column_name | cols] = out_df.names
+
+    correlations =
+      Enum.map(cols, fn left ->
+        corr_series =
+          cols
+          |> Enum.map(fn right -> PolarsSeries.correlation(df[left], df[right], ddof) end)
+          |> Shared.from_list({:f, 64})
+          |> Shared.create_series()
+
+        {left, corr_series}
+      end)
+
+    names_series = cols |> Shared.from_list(:string) |> Shared.create_series()
+
+    from_series([{column_name, names_series} | correlations])
+  end
```

Comment:
@josevalim we did using the backend because I wanted to raise for the lazy version, since we cannot implement for that right now. Do you think it worth to revert it anyway?

Reply:
This way it works with lazy and within `Explorer.DataFrame` as well! Something like:

```
mutate_with(df, fn ldf ->
  Enum.map(columns, ...)
end)
|> select(existing_columns -- new_columns, :discard)
```

WDYT?

Reply:
Oh, I'm going to try.

Reply:
@josevalim  I think this won't work the way we want, because `mutate_with` expects only lazy_series/expressions as values, and we are trying to create it with a list of lazy series. On the other hand, we could try create a column for each pair and try to pivot the results later. But again, pivoting does not work with lazy frames.

Maybe there is another way to reshape this DF, but I don't know yet.
I'm going to investigate more tomorrow :)

Reply:
Hrm, now that you mention it, I think you are right. Feel free to ship it. :)

Reply:
Actually, don't ship it yet. I would like Chris' approval on the API before. :D